apps/server/src/maps.ts

```diff
@@ -9,16 +19,30 @@ export async function geocodeAddress(address: string) {
     return null;
   }
 
-  return {
+  const location = {
     lat: body.features[0].geometry.coordinates[1],
     lng: body.features[0].geometry.coordinates[0],
   };
+
+  await redis.set(`geocode:${address}`, JSON.stringify(location), {
+    EX: 30 * 24 * 60 * 60,
+  });
+
+  return location;
 }
 
 export async function getDrivingDurationMinutes(
   p1: { lat: number; lng: number },
   p2: { lat: number; lng: number }
 ) {
+  const cachedDuration = await redis.get(
+    `drivingDuration:${p1.lat},${p1.lng}-${p2.lat},${p2.lng}`
```

Comment:
can we use .toFixed(3) on each number to reduce the coordinates to 3 decimal places? The coordinates might keep varying by a few meters here and there and that will keep re-caching in redis without a big difference in driving duration